adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/model/Datasource.java

```diff
@@ -78,6 +83,27 @@ public String toString() {
         '}';
     }
 
+    public AdapterConfig toAdapterConfig() {
+      return new AdapterConfig(
+        new ProtocolConfig(this.getProtocol().getType(), this.getProtocol().getParameters()),
+        new FormatConfig(this.getFormat().getType(), this.getFormat().getParameters())
+      );
+  }
+
+    public Datasource fillQueryParameters(DatasourceParameters datasourceParameters) {
+      if (datasourceParameters == null || datasourceParameters.parameters == null) {
+        return this;
+      }
+      String url = (String) this.getProtocol().getParameters().get("location");
```

Comment:
I think this logic should be protocol specific.. can we add that method to the Protocol Interface and move that logic to the HttpProtocol?

Reply:
The fillQueryParameter now returns a DatasourceProtocol and is used in the toAdapterConfig method. 
I added a check for Type "HTTP". 
Additional I added a test for the toAdapterConfig method.